### PR TITLE
Cut Version 0.65 Docs and Update API Documentation

### PR DESCRIPTION
**Pull request by chiaramooney**
**_What_**
Cut version 0.65 documentation for React Native Windows website. 

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/react-native-windows-samples/pull/471)

### CODE REVIEW

**Review by asklar — APPROVED** [review]
LGTM - when we are ready to flip the switch, we'll want to update website/siteConfig.js to point to 0.65 instead of 0.64

File: website/versioned_docs/version-0.65/customizing-SDK-versions.md [hunk 1]
```diff
@@ -0,0 +1,97 @@
+---
+id: version-0.65-customizing-sdk-versions
+title: Customizing SDK versions
+original_id: customizing-sdk-versions
+---
+
+In React Native for Windows 0.64.3, we have made it easier for an app to customize which versions of the Windows SDK and WinUI 2.x to use.
```
Comment (asklar) [inline website/versioned_docs/version-0.65/customizing-SDK-versions.md:7]: Maybe rephrase this now that this is for 65? "It is easy for an app to customize..." or something like that

**Review by asklar — APPROVED** [review]

**Comment by asklar** [comment]
looks like the calculator sample is failing to build with:
`error MSB3202: The project file "D:\a\src\node_modules\react-native-windows\JSI\Universal\JSI.Universal.vcxproj" was not found. `
@jonthysell ?

File: samples/Calculator/package.json [hunk 2]
```diff
@@ -12,8 +12,8 @@
   },
```
Comment (NickGerleman) [inline samples/Calculator/package.json:12]: @chiaramooney updating the samples will be a little bit more involved than changing just versions. I can paste over some instructions, but would you mind us doing that as a separate change?

**Comment by NickGerleman** [comment]
> looks like the calculator sample is failing to build with:
> `error MSB3202: The project file "D:\a\src\node_modules\react-native-windows\JSI\Universal\JSI.Universal.vcxproj" was not found. `
> @jonthysell ?

We will need to do an upgrade-helper style update of more dependencies, more build logic, etc. Docs for ACoates upgrade-helper version are in the preview release notes.

File: samples/Calculator/package.json (hunk 2 above)
Comment (chiaramooney) [reply samples/Calculator/package.json:12]: Yea definitely. Just did the version change because at first the Calc CI was failing during the `yarn install` but looks like it's break is more involved. Best we fix that in a separate PR.

**Comment by chiaramooney** [comment]
Looks like the website pipeline is claiming that it can't find this URL: "URL not found https://github.com/microsoft/WinAppDriver/releases/download/v1.1/WindowsApplicationDriver.msi while parsing versioned_docs/version-0.60/e2e-test.md (HTTP 403)"

The link downloads WinAppDriver for you and when I ran it just now it still works successfully. @jonthysell Any idea how to alter the link to make it pass the broken link checks or silence the error?

**Comment by jonthysell** [comment]
We shouldn't be updating the samples as part of the doc/website update. It should be a separate PR, and follow the instructions here: https://github.com/microsoft/react-native-windows-samples/tree/main/samples#upgrade-procedure

**Comment by jonthysell** [comment]
For example, here are the detailed steps to upgrade Calculator: https://github.com/microsoft/react-native-windows-samples/tree/main/samples/Calculator#upgrade

File: samples/Calculator/yarn.lock [hunk 3]
```diff
@@ -55,7 +55,7 @@
     semver "^6.3.0"
     source-map "^0.5.0"
 
-"@babel/core@^7.14.0":
```
Comment (jonthysell) [inline samples/Calculator/yarn.lock]: @chiaramooney  Don't forget to revert this change too.

**Comment by chiaramooney** [comment]
@jonthysell Oop just caught the mistake; that version update for Calculator was not intentional. Just removed that change now.
